niche_vlaanderen/acidity.py

```diff
@@ -59,48 +75,88 @@ def __init__(
+        soil_group = self._ct_soil_codes.soil_group[soil_code].values.astype("uint8")
 
+        result = np.ma.empty_like(soil_code)
+        for sel_group, subtable in self._ct_soil_mlw.groupby("soil_group"):
             subtable = subtable.copy().reset_index(drop=True)
             index = np.digitize(mlw, subtable.mlw_max, right=False)
+            index[nodata.flatten()] = self.nodata
             selection = soil_group == sel_group
-            result[selection] = subtable.soil_mlw_class.reindex(index)[selection]
-
-        result[(mlw == 99) | (np.isnan(mlw))] = -99
-        result = result.reshape(orig_shape)
+            subtable.loc[self.nodata, "soil_mlw_class"] = self.nodata
+            result[selection] = subtable.soil_mlw_class.reindex(index).iloc[selection].values
+        result = result.reshape(orig_shape).astype(self.dtype)
+        result[nodata] = self.nodata  # Apply the nodata mask
         return result
 
-    def _get_acidity(self, rainwater, minerality, inundation, seepage, soil_mlw_class):
+    def _get_acidity(self, rainwater, minerality, inundation,
+                     seepage_class, soil_mlw_class):
+        """Calculate the acidity
 
+        Parameters
+        ----------
+        rainwater : numpy.ndarray, numpy.uint8
+            Array denoting whether rainwater lenses occur.
+        minerality : numpy.ndarray, numpy.uint8
+            Array noting whether the groundwater is rich(1) or poor in minerals
+        inundation : numpy.ndarray, numpy.uint8
+            Array containing rate of inundation.
+        seepage_class : numpy.ndarray, numpy.uint8
+            Array containing the classified version of theflux of groundwater.
+        soil_mlw_class : numpy.ndarray, numpy.uint8
+            Array containing the soil mlw classes from helper function.
+
+        Returns
+        -------
+        numpy.ndarray, numpy.uint8
+            Array containing the acidity values.
+        """
         orig_shape = inundation.shape
 
         check_codes_used("rainwater", rainwater, {0, 1})
         check_codes_used(
             "minerality", minerality, self._lnk_acidity["mineral_richness"]
         )
         check_codes_used("inundation", inundation, self._lnk_acidity["inundation"])
-        check_codes_used("seepage", seepage, self._ct_seepage["seepage"])
+        check_codes_used("seepage", seepage_class, self._ct_seepage["seepage"])
+
+        nodata = ((rainwater == 255) | (minerality == 255) |
+                  (inundation == 255) | (soil_mlw_class == 255) |
+                  (seepage_class == 255))
 
         rainwater = rainwater.flatten()
         minerality = minerality.flatten()
         inundation = inundation.flatten()
-        seepage = seepage.flatten()
+        seepage = seepage_class.flatten()
         soil_mlw_class = soil_mlw_class.flatten()
 
-        result = np.full(soil_mlw_class.shape, self.nodata, dtype="uint8")
+        result = np.empty_like(soil_mlw_class)
```

Comment:
np.empty does not initialize entries. I think it is safe because we check our input values, but defaulting to nodata seems a bit safer in case a combination is not matched.

Reply:
I opt for `np.ones() *self.nodata` instead, result should be the same